service/s3/s3manager/cleanup.go

```diff
@@ -0,0 +1,5 @@
+package s3manager
+
+// CleanupFunc is used to signal that resources associated with the function
+// may be reaped for reuse.
+type CleanupFunc func()
```

Comment:
I like that approach as well, wasn't sure from a stylistic perspective if we liked to utilize named parameters. It seems pretty heavy handed to have to export this specific type otherwise.